src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSourceConfig.java

```diff
@@ -173,6 +199,27 @@ public void validate() {
       if (null == fields || fields.isEmpty()) {
         throw new InvalidConfigPropertyException("Schema should contain fields to map", SCHEMA);
       }
+      if (columnMappings != null) {
+        Set<String> mappedFieldNames = Sets.newHashSet(columnMappings.values());
+        Set<String> schemaFieldNames = fields.stream()
+          .map(Schema.Field::getName)
+          .filter(name -> !name.equals(keyAlias))
+          .collect(Collectors.toSet());
+        Sets.SetView<String> nonDescribedFields = Sets.difference(schemaFieldNames, mappedFieldNames);
+        if (!nonDescribedFields.isEmpty()) {
+          String nonDescribedFieldsString = String.join(", ", nonDescribedFields);
+          String errorMessage = String.format("Some mapped columns are not described in schema: %s. ",
+                                              nonDescribedFieldsString);
+          throw new InvalidConfigPropertyException(errorMessage, SCHEMA);
+        }
+        Sets.SetView<String> nonMappedColumns = Sets.difference(mappedFieldNames, schemaFieldNames);
+        if (!nonMappedColumns.isEmpty()) {
+          String nonMappedColumnsString = String.join(", ", nonDescribedFields);
+          String errorMessage = String.format("Some schema fields do not have corresponding column mappings: %s. ",
```

Comment:
I can hardly imagine the use case when it was intended by the user. Are you sure I should replace it with warning? User may miss some fields.

Reply:
In general, I think it's better to try not to make assumptions about user intent, as some users will end up trying to use it differently than we think.

For example, this can happen if the mapping is a macro that is a common value shared across multiple pipelines. The value could be a superset of everything needed, in order to managing one property per pipeline. Or maybe they want to debug a pipeline so they have cloned it and they want to delete a bunch of fields from the output schema, but leave the mapping because they intend to add most of them back after debugging.

Reply:
this still needs to be addressed.